Watchman.Discord/Areas/Users/Controllers/WarnsController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Devscord.DiscordFramework.Commons;
+using Devscord.DiscordFramework.Commons.Exceptions;
+using Devscord.DiscordFramework.Framework.Architecture.Controllers;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using Devscord.DiscordFramework.Framework.Commands.Responses;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using Devscord.DiscordFramework.Services.Factories;
+using MongoDB.Bson;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Discord.Areas.Protection.Services;
+using Watchman.Discord.Areas.Users.BotCommands;
+using Watchman.Discord.Areas.Users.BotCommands.Warns;
+using Watchman.DomainModel.Messages;
+using Watchman.DomainModel.Users;
+
+namespace Watchman.Discord.Areas.Protection.Controllers
+{
+    public class WarnsController : IController
+    {
+        private readonly MessagesServiceFactory _messagesServiceFactory;
+        private readonly UsersService _usersService;
+        private readonly DirectMessagesService _directMessagesService;
+        private readonly WarnService _warnService;
+
+        public WarnsController(MessagesServiceFactory messagesServiceFactory, UsersService usersService, DirectMessagesService directMessagesService, WarnService warnService)
+        {
+            this._messagesServiceFactory = messagesServiceFactory;
+            this._usersService = usersService;
+            this._directMessagesService = directMessagesService;
+            this._warnService = warnService;
+        }
+
+        [AdminCommand]
+        public async Task AddWarn(AddWarnCommand command, Contexts contexts)
+        {
+            var messageService = _messagesServiceFactory.Create(contexts);
+            var mentionedUser = _usersService.GetUserByMention(contexts.Server, command.User.ToString());
```

Comment:
this.

Reply:
czy "this" nie jest równoznaczne z _ w nazwie?